include/bluetooth/services/ots_srv.h

```diff
@@ -0,0 +1,641 @@
+			     uint32_t offset);
+};
+
+/** @brief Descriptor for OTS initialization. */
+struct bt_gatt_ots_srv_init {
+	/* OTS features */
+	struct bt_gatt_ots_srv_feat features;
+
+	/* Callbacks */
+	struct bt_gatt_ots_srv_cb *cb;
+};
+
+/** @brief Add an object to the OTS Server.
+ *
+ *  This function adds an object to the OTS database. When the
+ *  object is being added, a callback obj_created() is called
+ *  to notify the user about a new object ID.
+ *
+ *  @param ots      OTS Server instance.
+ *  @param obj_init Meta data of the object.
+ *
+ *  @return 0 in case of success or negative value in case of error.
+ */
+int bt_gatt_ots_srv_object_add(struct bt_gatt_ots_srv *ots,
+			       struct bt_gatt_ots_srv_obj_metadata *obj_init);
+
+/** @brief Delete an object from the OTS Server.
+ *
+ *  This function deletes an object from the OTS database. When the
+ *  object is deleted a callback obj_deleted() is called
+ *  to notify the user about this event. At this point, it is possible
+ *  to free allocated buffer for object data.
+ *
+ *  @param ots OTS Server instance.
+ *  @param id  ID of the object to be deleted (uint48).
+ *
+ *  @return 0 in case of success or negative value in case of error.
+ */
+int bt_gatt_ots_srv_object_delete(struct bt_gatt_ots_srv *ots, uint64_t id);
+
+/** @brief Get the service declaration attribute.
+ *
+ *  This function is enabled for BT_GATT_OTS_SRV_SECONDARY_SVC configuration.
+ *  The first service attribute can be included in any other GATT service.
+ *
+ *  @param ots OTS Server instance.
+ *
+ *  @return The first OTS attribute instance.
+ */
+void *bt_gatt_ots_srv_svc_decl_get(struct bt_gatt_ots_srv *ots);
+
+/** @brief Initialize the OTS Server instance.
+ *
+ *  @param ots      OTS Server instance.
+ *  @param ots_init OTS initialization descriptor.
+ *
+ *  @return 0 in case of success or negative value in case of error.
+ */
+int bt_gatt_ots_srv_init(struct bt_gatt_ots_srv *ots,
+			 struct bt_gatt_ots_srv_init *ots_init);
+
+/** @brief Get a free instance of OTS Server from the pool.
+ *
+ *  @return OTS instance in case of success or NULL in case of error.
+ */
+struct bt_gatt_ots_srv *bt_gatt_ots_srv_free_instance_get(void);
+
+/** @brief Converts binary OTS Object ID to string.
+ *
+ *  @param object_id Object ID.
+ *  @param str       Address of user buffer with enough room to store
+ *                   formatted string containing binary Object ID.
+ *  @param len       Length of data to be copied to user string buffer.
+ *                   Refer to BT_GATT_OTS_SRV_OBJECT_ID_STR_LEN about
+ *                   recommended value.
+ *
+ *  @return Number of successfully formatted bytes from binary ID.
+ */
+static inline int bt_gatt_ots_srv_object_id_to_str(uint64_t object_id,
+						   char *str, size_t len)
+{
+	uint8_t id[6];
+
+	sys_put_le48(object_id, id);
+
+	return snprintk(str, len, "0x%02X%02X%02X%02X%02X%02X",
+			id[5], id[4], id[3], id[2], id[1], id[0]);
+}
```

Comment:
I would leave it for now, it could be refactored in future PRs without breaking API since it's a new function.

BTW, it seems to me that ``bt_addr_le_str_real`` is not available as public API.

Reply:
> BTW, it seems to me that bt_addr_le_str_real is not available as public API.
That is true (been an annoyance for me when I implement my test applications :P).  

This is indeed more similar to the public string functions such as `bt_uuid_to_str`.